### PR TITLE
feature/446-check-registration-steps

### PR DESCRIPTION
**Pull request by HaseebUllahAbbasi**
user Registration process with cases defined in issue

### CODE REVIEW

File: src/store/userStore.ts [hunk 1]
```diff
@@ -5,8 +5,22 @@ import settings from '../settings';
 import { User, UserStatus, createUserObject, setSettings } from 'tonomy-id-sdk';
 
 // TODO change this to be an instance of User class when we have implemented the RNKeyStore
+// type UserStats ={
+
+// }
+type statusData = 'isLoggedIn' | 'creatingAccount' | 'NotLoggedIn' | 'LoggingIn';
```
Comment (Rebal67) [inline src/store/userStore.ts:11]: lets move this to enum for faster value changes

File: src/store/userStore.ts [hunk 2]
```diff
@@ -5,8 +5,22 @@ import settings from '../settings';
...
     user: User;
+    hasCompletedPin: boolean;
+    hasCompletedBiometric: boolean;
+    status: statusData;
+    getStatus: () => statusData;
+    setStatus: (statusData) => void;
+    checkPin: () => boolean;
+    updatePin: (pin) => void;
+    checkBiometric: () => boolean;
+    updateBiometric: (pin) => void;
```
Comment (Rebal67) [inline src/store/userStore.ts:22]: isn't status and get status doing the same thing ?

File: src/store/userStore.ts [hunk 3]
```diff
@@ -19,6 +33,26 @@ setSettings({
...
+
+    getStatus: () => {
+        return get().status;
+    },
+    updatePin: () => {
+        set((state) => ({ ...state, hasCompletedPin: true }));
+    },
+    updateBiometric: () => {
+        set((state) => ({ ...state, hasCompletedBiometric: true }));
+    },
```
Comment (Rebal67) [inline src/store/userStore.ts:54]: why are you spreading the state what will this do?
you can just use set({hasCompletedBiometric: true })

Comment (Rebal67) [reply src/store/userStore.ts:54]: i am not seeing presistant storage here not the ability to set the flags on false

File: src/containers/LoginPasswordContainer.tsx [hunk 4]
```diff
@@ -27,18 +27,27 @@ export default function LoginPasswordContainer({
...
-            });
+
+            if (!store.checkPin())
+                navigation.navigate('CreateAccountPin', {
+                    password: password,
+                });
+            else {
+                if (!store.checkBiometric()) {
+                    navigation.navigate('CreateAccountFingerprint', {
+                        password: password,
```
Comment (Rebal67) [inline src/containers/LoginPasswordContainer.tsx:47]: can you use block for the if statement here

Comment (Rebal67) [reply src/containers/LoginPasswordContainer.tsx:47]: shouldnt status be updated here to user is loggedin

**Review by Rebal67 — CHANGES_REQUESTED** [review]
please check the DOD of this task, this is definitely not finished

File: src/store/userStore.ts [hunk 5]
```diff
@@ -3,11 +3,32 @@ import RNKeyManager from '../utils/RNKeyManager';
...
+export enum StatusData {
+    IS_LOGGED_IN,
+    CREATING_ACCOUNT,
+    NOT_LOGGED_IN,
+    LOGGING_IN,
+}
+
 interface UserState {
     user: User;
+    hasCompletedPin: boolean;
```
Comment (theblockstalk) [inline src/store/userStore.ts:21]: hasCompletedPin and hasCompletedBiometric are not implemented. remove

File: src/store/userStore.ts [hunk 6]
```diff
@@ -3,11 +3,32 @@ import RNKeyManager from '../utils/RNKeyManager';
...
+    hasCompletedBiometric: boolean;
+    status: StatusData;
+    setStatus: (statusData) => void;
+    checkPin: () => boolean;
```
Comment (theblockstalk) [inline src/store/userStore.ts:25]: checkPin rename getPin
checkBiometric rename getBiometric

updatePin and Biometric do the same thing logical as setPin and Biometric, they can be removed

File: src/containers/PinScreenContainer.tsx [hunk 7]
```diff
@@ -39,15 +39,20 @@ export default function PinScreenContainer({
...
-                    await user.savePIN(confirmPin);
-                } catch (e) {
-                    console.log('error saving pin', e);
+                    await store.user.savePIN(confirmPin);
+                    await store.updatePin(true);
+
+                    if (!(await store.checkBiometric())) {
+                        navigation.navigate('CreateAccountFingerprint', { password });
+                    } else {
+                        await store.removeFlags();
```
Comment (theblockstalk) [inline src/containers/PinScreenContainer.tsx:48]: remove flags should be at the end of biometrics I think

File: src/containers/PinScreenContainer.tsx [hunk 8]
```diff
@@ -39,15 +39,20 @@ export default function PinScreenContainer({
         if (confirming) {
             if (pin === confirmPin) {
                 try {
-                    await user.savePIN(confirmPin);
-                } catch (e) {
-                    console.log('error saving pin', e);
+                    await store.user.savePIN(confirmPin);
+                    await store.updatePin(true);
+
+                    if (!(await store.checkBiometric())) {
```
Comment (theblockstalk) [inline src/containers/PinScreenContainer.tsx:45]: this is never true, they have never finished Biometric before PIN

File: src/containers/LoginPasswordContainer.tsx [hunk 9]
```diff
@@ -27,18 +27,27 @@ export default function LoginPasswordContainer({
...
-            navigation.navigate('CreateAccountPin', {
-                password: password,
-            });
+
+            if (!(await store.checkPin())) {
+                navigation.navigate('CreateAccountPin', {
+                    password: password,
+                });
+            } else {
+                if (!(await store.checkBiometric())) {
```
Comment (theblockstalk) [inline src/containers/LoginPasswordContainer.tsx:45]: user always navigated to PIN after they login

Comment (theblockstalk) [reply src/containers/LoginPasswordContainer.tsx:45]: no need to check Biometric or Pin here

File: src/store/userStore.ts [hunk 10]
```diff
@@ -3,11 +3,32 @@ import RNKeyManager from '../utils/RNKeyManager';
 import { storageFactory } from '../utils/storage';
 import settings from '../settings';
 import { User, UserStatus, createUserObject, setSettings } from 'tonomy-id-sdk';
+import AsynStorage from '@react-native-async-storage/async-storage';
 
 // TODO change this to be an instance of User class when we have implemented the RNKeyStore
+// type UserStats ={
```
[the remark this answers is not in the document]
Comment (HaseebUllahAbbasi) [reply src/store/userStore.ts:9]: sure

File: src/store/userStore.ts [hunk 11]
```diff
@@ -19,14 +40,47 @@ setSettings({
...
+    },
     isLoggedIn: async () => {
         const status = await get().user.storage.status;
 
         return status && status === UserStatus.READY;
     },
-
     logout: async () => {
-        await get().user.logout();
+        await set().user.logout();
```
[the remark this answers is not in the document]
Comment (HaseebUllahAbbasi) [reply src/store/userStore.ts:83]: yeah it is get

File: src/store/userStore.ts (hunk 5 above)
Comment (HaseebUllahAbbasi) [reply src/store/userStore.ts:21]: removing

File: src/store/userStore.ts (hunk 6 above)
Comment (HaseebUllahAbbasi) [reply src/store/userStore.ts:25]: sure

File: src/containers/LoginPasswordContainer.tsx (hunk 9 above)
Comment (HaseebUllahAbbasi) [reply src/containers/LoginPasswordContainer.tsx]: removed

**Pull request closed without merging**